app/models/foreman_fog_proxmox/proxmox_vm_commands.rb

```diff
@@ -31,11 +31,13 @@ def start_on_boot(vm, args)
       vm
     end
 
+    # rubocop:disable Metrics/AbcSize
     def create_vm(args = {})
       vmid = args[:vmid].to_i
       type = args[:type]
       node = client.nodes.get(args[:node_id])
       vmid = node.servers.next_id.to_i if vmid < 1
+      args[:vmid] = vmid.to_s if args[:vmid].nil?
```

Comment:
Can you please explain? This looks strange - as in line 35 the vmid is set to `args[:vmid].to_i`.

to reduce the AbcSize, we may put everything to get/calculate the vmid in a separte method?

Reply:
In line 35, we set vmid to args[:vmid] for clone_from_image part which is on line 46. But for API, args[:vmid] can be nil that's why it needs to be assigned by vmid which is set on line 39. vmid on line is only set if user send arg[:vmid] to 0 or leave it nil. 

We need args[:vmid] to have some value to create vm in line 56.

to reduce the AbcSize, we need to set both vmid and args[:vmid] if they are < 1. Even if we call different function. still there will be one/two assignments depending on condition (it's not necessary args[:vmid] is nil but still vmid < 1)

Reply:
Thanks for the explanation. Can we somehow improve this code to make it more readable. You know the magic behind but nobody would understand it from the written code.